feat: add ContainerResource type for HPA cpu/memory scaler (#876)
kedacore/keda-docs: content/docs/2.8/scalers/cpu.md

- `value` - Value to trigger scaling actions for:
	- When using `Utilization`, the target value is the average of the resource metric across all relevant pods, represented as a percentage of the requested value of the resource for the pods.
	- When using `AverageValue`, the target value is the target value of the average of the metric across all relevant pods (quantity).

> - `containerName` - Set this to target the specific container rather than the entire pod. Defaults to empty if not specified.

Reviewer: ```suggestion
- `containerName` - Name of the specific container to scaled based on, rather than the entire pod. Defaults to empty if not specified.
```

Reviewer 2: `to scaled based on` ?

Reviewer: `to scale on` was more what I meant or ´to scale based on its CPU` maybe?

Reviewer 2: `to scale based on its CPU` is good.

Reviewer 2: and the same for memory docs

Reviewer: suggestion updated

    metricType: Utilization
    metadata:
      value: "50"

>     containerName: "foo"

Reviewer: Let's use decent names
```suggestion
    containerName: "api"
```
